plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -865,6 +868,103 @@ public void testTextualPredicatePushdown()
                 .isFullyPushedDown();
     }
 
+    @Test
+    public void testOrPredicatePushdown()
+    {
+        assertThat(query("SELECT * FROM nation WHERE name = 'ALGERIA' OR comment = ''")).isFullyPushedDown();
+        assertThat(query("SELECT * FROM nation WHERE name IS NULL OR comment IS NULL")).isFullyPushedDown();
+    }
+
+    @Test
+    public void testLikePredicatePushdown()
+    {
+        assertThat(query("SELECT nationkey FROM nation WHERE name LIKE '%A%'"))
+                .isFullyPushedDown();
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_like_predicate_pushdown",
+                // TODO Move to a bounded varchar of length 1. https://starburstdata.atlassian.net/browse/SEP-11316
+                "(id integer, a_varchar varchar)",
+                List.of(
+                        "1, 'A'",
+                        "2, 'a'",
+                        "3, 'B'",
+                        "4, 'ą'",
+                        "5, 'Ą'"))) {
+            assertThat(query("SELECT id FROM " + table.getName() + " WHERE a_varchar LIKE '%A%'"))
+                    .isFullyPushedDown();
+            assertThat(query("SELECT id FROM " + table.getName() + " WHERE a_varchar LIKE '%ą%'"))
+                    .isFullyPushedDown();
+        }
+    }
+
+    @Test
+    public void testIsNullPredicatePushdown()
+    {
+        assertThat(query("SELECT nationkey FROM nation WHERE name IS NULL")).isFullyPushedDown();
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_is_null_predicate_pushdown",
+                "(a_int integer, a_varchar varchar(1))",
+                List.of(
+                        "1, 'A'",
+                        "2, 'B'",
+                        "1, NULL",
+                        "2, NULL"))) {
+            assertThat(query("SELECT a_int FROM " + table.getName() + " WHERE a_varchar IS NULL")).isFullyPushedDown();
```

Comment:
this is probably being handled by tuple domain since `IS NULL` is not a connector expression.

Reply:
Good catch. 
This behaviour appeared in this PR because of not intended simplification of https://github.com/trinodb/trino/pull/22921
Unfortunately I don't see a reliable way to enforce usage of connector expression, thus OR has to be used.
https://github.com/trinodb/trino/pull/23455#issuecomment-2355788346
> > connector expression pushdown or tuple domain
> 
> I guess for simple expression we use TupleDomain and for expression with OR it would use connector expression